Add soil temperature soil moisture

Justin-Mann opened the pull request:
## Thank you for sending in a pull request, here's some tips to get started!

(Please delete all these tips and replace with your text)

- Before starting on some new big chunk of code, it it is optional but highly recommended to open an issue first to say "Hey, I think this idea X should be implemented and I'm starting work on it. My general plan is Y, any feedback is appreciated." This will allow other devs to potentially save you time by not accidentally duplicating work etc...
- Mention "#(issue)" in the description, when applicable
- Please do not check in files that don't have real changes
- Please do not reformat lines that you didn't have to change the code on
- If your other co-developers have comments on your PR please tweak as needed
- Do not use any external image service, just paste or drag and drop the image here and it will be uploaded automatically
- Please also enable "Allow edits by maintainers".

<!--
If you have screenshots or recordings to display your change, please include them!

You can use this template for displaying a single screenshot:
<img src="" width="300"/>

or a video recording:
<video src="" width="300"></video>


And if you want to display the state before and after a change, you can use this table template:

| Before | After |
|------|-----|
| <img src="" width="300"/> | <img src="" width="300"/> |
-->

DaneEvans reviewed (commented):
lgtm when it's ready

jamesarich commented:
@Justin-Mann what's your intention here? Are you going to do any more work on this PR?

Justin-Mann commented:
@jamesarich is here for u 2 advise man. :) i will follow up on your advise, but will be a minute before I have additional time to understand kotlin. Anyone can pick this up and carry it forward and I would not be offended.

jamesarich commented:
So - this PR as it stands will start saving these sensor packets to the app's packet log, which could be then viewed/exported from the Debug Panel.
Once passing detekt checks - We could merge this PR in, and it would be a good first step for someone else to take on implementation of a specific log and chart for these sensors if so desired.

Justin-Mann commented:
@jamesarich thnx man. I'll work on it a Lil more and get it there.. the end goal for me is to display these in the app as the temperature and humidity metrics are displayed.  We are using this irl as part of a irrigation monitoring solution. I currently have a firmware build that cheats these values into the android app by writing them to the temperature and humidity properties. I'm headed home and will do a Lil work once there.

Justin-Mann commented:
@DaneEvans no love... 

am I ok to use custom?  soil Temp and soil moisture are the goals... 
![image](https://github.com/user-attachments/assets/b463b6ab-e133-4302-a845-99a7a6346aa7)

Justin-Mann commented:
@DaneEvans I think I did the thing man? having some issues actually testing on my device, but I got it to run a few times and it seems to be picking up the new metrics (soil_temperature and soil_moisture).  wondering if you can take a look-see and let me know if I'm deficient in the codebase or any CI/CD. typically I stay far away from Android studio.

Justin-Mann commented:
@DaneEvans lol sry for spamming you with msgs.. wrong dude obviously.. think this is close to rdy fyi..

Justin-Mann commented:
Do you need me to remove them? (Please no) ... is on me if so.. I kinda knew that shiz had to be automated.

DaneEvans commented:
James, is that written in the readme? Because that's 2 of us in a row that
have done that for our first MR.

If the strings files can handle it, a comment at the top would probably do
it too

On Tue, 1 July 2025, 9:47 am James Rich, ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> We only need the english strings file - the rest will be done by crowdin :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshtastic/Meshtastic-Android/pull/2227#pullrequestreview-2973079785>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIRN4N3QM3ZBOA4MNGYUYD3GHEAHAVCNFSM6AAAAAB73V7HWGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSNZTGA3TSNZYGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

DaneEvans commented on app/src/main/java/com/geeksville/mesh/model/Node.kt:
```diff
@@ -121,13 +121,26 @@ data class Node(
         }
         val humidity = if (relativeHumidity != 0f) "%.0f%%".format(relativeHumidity) else null
+        val soilTemp = if (soilTemperature != 0f) {
+            if (isFahrenheit) {
```
Do we really not have a real to yankee converter function? 
Given how many places it is used, it really should be one utility, not written in each spot. 

But it this is how it is in the other places, I'll allow it. Especially as they apparently aren't working at the moment anyway.

Justin-Mann commented:
Did another thing. Coulda probably done it better and more completely if u guys want it.

Justin-Mann commented on app/src/main/java/com/geeksville/mesh/model/MetricsViewModel.kt:
```diff
@@ -305,8 +305,10 @@ class MetricsViewModel @Inject constructor(
                         environmentMetrics = telemetry.filter {
                             it.hasEnvironmentMetrics() &&
-                                    it.environmentMetrics.relativeHumidity >= 0f &&
-                                    !it.environmentMetrics.temperature.isNaN()
+                                    (it.environmentMetrics.relativeHumidity >= 0f &&
+                                    !it.environmentMetrics.temperature.isNaN()) ||
+                                    (it.environmentMetrics.soilMoisture >= 0 &&
+                                    !it.environmentMetrics.soilTemperature.isNaN())
```
oops.. I may have committed that. I look and think. do what is intended.

Justin-Mann replied:
pretty sure we good. ty for review.

jamesarich reviewed (approved):
This checks out codewise, but I have no sensors to test the functionality.

DaneEvans commented on app/src/main/java/com/geeksville/mesh/android/TelemetryUtils.kt:
```diff
@@ -0,0 +1,27 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.geeksville.mesh.android
+
+/**
+ * Created by jem on 07/01/2025
+ */
+object TelemetryUtils {
+    fun toFahrenheit(tempCelsius: Float): Float {
+        return (tempCelsius * 1.8F) + 32
+    }
+}
```
```suggestion
package com.geeksville.mesh.android

private const val FARENHEIT_MULTIPLIER = 1.8F
private const val FARENHEIT_OFFSET = 32

object TelemetryUtils {
    fun toFahrenheit(tempCelsius: Float): Float {
        return (tempCelsius * FARENHEIT_MULTIPLIER) + FARENHEIT_OFFSET
    }
}
```

DaneEvans commented on app/src/main/java/com/geeksville/mesh/model/EnvironmentMetricsState.kt:
```diff
@@ -107,6 +118,27 @@ data class EnvironmentMetricsState(
+            telemetries.minBy { it.environmentMetrics.soilTemperature },
+            telemetries.maxBy { it.environmentMetrics.soilTemperature }
+        )
+        if (minSoilTemp.environmentMetrics.soilTemperature != 0f || maxSoilTemp.environmentMetrics.soilTemperature != 0f) {
```
```suggestion
        if (minSoilTemp.environmentMetrics.soilTemperature != 0f ||
            maxSoilTemp.environmentMetrics.soilTemperature != 0f) {
```

DaneEvans commented on app/src/main/java/com/geeksville/mesh/model/EnvironmentMetricsState.kt:
and add a , CyclomaticComplexMethod to line 88. 

That should be enough to keep detekt happy. 

The filtered function should probably get a refactor at some point, but it needent be now

Justin-Mann commented:
@jamesarich I do.

https://www.instructables.com/editor/ESUUPXOM5JMIYCY

I wrote the firmware submission.  Also believe @thebentern may have checked the functionality with the RAK Wisblock stack that I use before the firmware. Happy to resolve the other codestandards stuff tomorrow.

DaneEvans commented on app/src/main/java/com/geeksville/mesh/model/EnvironmentMetricsState.kt:
```diff
@@ -73,6 +84,7 @@ data class EnvironmentMetricsState(
      *
      * @param timeFrame used to filter
      * @return [EnvironmentGraphingData]
+     * CyclomaticComplexMethod
      */
     @Suppress("LongMethod")
```
```suggestion
      * @return [EnvironmentGraphingData]
      */
     @Suppress("LongMethod","CyclomaticComplexMethod")
```

Justin-Mann commented:
@jamesarich please add yourself as co-author if you like.

Justin-Mann commented:
last thing... just occurred to me.. 
I DO NOT run a regular temp/humidity monitor on my RAK device atm. 
so..
not tested in tandem AND the temp/humidity, soiltemp/soilmoisture strings on the infocards are gonna be very confusing when running them together (which is which type thing) node infocards.

I'm off unless you want me to do a thing.. I get out the way and let the good work happen.

Justin-Mann commented:
@jamesarich @DaneEvans 
Fml.. struggles this am.. it is my intention that everyone on the thread be added as coauthors if you as individuals want. Do what ya need to do to make it happen.

jamesarich commented:
We'll get to it as we can @Justin-Mann 👍

Justin-Mann commented:
No worries. Just trying to put a pin in it on my end. Metrics string (soil temp/moisture vs ambient temp/moisture) bothers me a bit, but I just gonna accept it. :) for now.

Justin-Mann commented:
Noticed a thing running the latest android build.. I know its not the place for it, but in between naps. Lol.. where im at with things today.. phone app crashes if I try to pull up environmental telemetry logs.

Justin-Mann commented:
@DaneEvans ya.. still happening.  everything on firmware 2.7.1.(latest) Alpha

Justin-Mann commented:
@DaneEvans there ya go.  happy to do more evasive firmware flash (ie - "full erase and install") or whatevs.. I should probably not be leaning into the naps anyways. tired 2day.

DaneEvans commented:
````
2025-07-04 17:22:43.684 12846-12846 AndroidRuntime com.geeksville.mesh E FATAL EXCEPTION: main
Process: com.geeksville.mesh, PID: 12846
java.util.NoSuchElementException
at com.geeksville.mesh.model.EnvironmentMetricsState.environmentMetricsFiltered(Unknown Source:1111)
at com.geeksville.mesh.ui.metrics.EnvironmentMetricsKt.EnvironmentMetricsScreen(Unknown Source:178)
```

No such element. 

Did you say you're on an app store version? This MR shouldn't be in there yet, so it definitely won't show, but it also shouldn't crash.

Justin-Mann commented:
correct.  If I was dinking with the branch build I would not have posted.

Justin-Mann commented:
also occurs if I bt directly to the RAK and try to look @ the e. log

Justin-Mann commented:
can connect to RAK and pull logs if helpful.

Justin-Mann commented:
i fire up vscode now, platformio takes a minute.

Justin-Mann commented:
no noteable activity on RAK when I try to look at log. 

--- Terminal on COM19 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_c3_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
I**NFO  | ??:??:?? 19 [EnvironmentTelemetry] Init sensor: RAK12035
INFO  | ??:??:?? 19 [EnvironmentTelemetry] RAK12035Sensor Init Succeed 
Sensor1 Firmware version: 2, Sensor Name: RAK12035
INFO  | ??:??:?? 19 [EnvironmentTelemetry] Opened RAK12035 sensor on i2c bus
INFO  | ??:??:?? 21 [EnvironmentTelemetry] Dry calibration value is 505
INFO  | ??:??:?? 21 [EnvironmentTelemetry] Wet calibration value is 382**
INFO  | ??:??:?? 21 [PowerTelemetry] Power Telemetry: init
DEBUG | ??:??:?? 25 [Power] Battery: usbPower=1, isCharging=1, batMv=3975, batPct=78
INFO  | ??:??:?? 35 [NodeInfo] Send our nodeinfo to mesh (wantReplies=0)
INFO  | ??:??:?? 35 [NodeInfo] Send owner !f5032f44/Water Monitor 6/WM6
DEBUG | ??:??:?? 35 [NodeInfo] Initial packet id 2056025763
DEBUG | ??:??:?? 35 [NodeInfo] Partially randomized packet id 3599438500
DEBUG | ??:??:?? 35 [NodeInfo] Update DB node 0xf5032f44, rx_time=0
INFO  | ??:??:?? 35 [NodeInfo] Sort took 0 milliseconds
DEBUG | ??:??:?? 35 [NodeInfo] handleReceived(LOCAL) (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x0 Portnum=4 priority=10)
DEBUG | ??:??:?? 35 [NodeInfo] No modules interested in portnum=4, src=LOCAL
INFO  | ??:??:?? 35 [NodeInfo] Packet History - insert: Using new slot @uptime 35.439s TRACE NEW
DEBUG | ??:??:?? 35 [NodeInfo] Use AES256 key!
DEBUG | ??:??:?? 35 [NodeInfo] enqueue for send (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=103 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 35 [NodeInfo] txGood=0,txRelay=0,rxGood=0,rxBad=0
WARN  | ??:??:?? 35 [RadioIf] Can not send yet, busyRx
WARN  | ??:??:?? 35 [RadioIf] Can not send yet, busyRx
WARN  | ??:??:?? 35 [RadioIf] Can not send yet, busyRx
WARN  | ??:??:?? 35 [RadioIf] Can not send yet, busyRx
WARN  | ??:??:?? 35 [RadioIf] Can not send yet, busyRx
DEBUG | ??:??:?? 35 [RadioIf] Ignore false preamble detection
DEBUG | ??:??:?? 36 [RadioIf] Started Tx (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=103 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 36 [RadioIf] Packet TX: 1001ms
DEBUG | ??:??:?? 36 [RadioIf] 0 packets remain in the TX queue
DEBUG | ??:??:?? 37 [RadioIf] Completed sending (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=103 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 38 [RadioIf] Lora RX (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=103 rxSNR=6.5 rxRSSI=-53 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 38 [RadioIf] Packet RX: 1001ms
DEBUG | ??:??:?? 38 [Router] Rx someone rebroadcasting for us (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=103 rxSNR=6.5 rxRSSI=-53 hopStart=6 rel
DEBUG | ??:??:?? 38 [Router] Didn't find pending packet
INFO  | ??:??:?? 38 [Router] Packet History - insert: Reusing slot aged 2.726s TRACE MATCHED PACKET
DEBUG | ??:??:?? 38 [Router] Ignore dupe incoming msg (id=0xd68b12a4 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=103 rxSNR=6.5 rxRSSI=-53 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 38 [Router] cancelSending id=0xd68b12a4, removed=0
DEBUG | ??:??:?? 38 [Router] Incoming msg was filtered from 0xf5032f44
DEBUG | ??:??:?? 45 [Power] Battery: usbPower=1, isCharging=1, batMv=3981, batPct=79
INFO  | ??:??:?? 65 [DeviceTelemetry] Send: air_util_tx=0.027806, channel_utilization=3.336667, battery_level=101, voltage=3.981000, uptime=65
DEBUG | ??:??:?? 65 [DeviceTelemetry] Partially randomized packet id 2523221669
DEBUG | ??:??:?? 65 [DeviceTelemetry] updateTelemetry LOCAL
DEBUG | ??:??:?? 65 [DeviceTelemetry] Node status update: 3 online, 3 total
INFO  | ??:??:?? 65 [DeviceTelemetry] Send packet to mesh
DEBUG | ??:??:?? 65 [DeviceTelemetry] Update DB node 0xf5032f44, rx_time=0
INFO  | ??:??:?? 65 [DeviceTelemetry] Sort took 0 milliseconds
DEBUG | ??:??:?? 65 [DeviceTelemetry] handleReceived(LOCAL) (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x0 Portnum=67 priority=10)
DEBUG | ??:??:?? 65 [DeviceTelemetry] No modules interested in portnum=67, src=LOCAL
INFO  | ??:??:?? 65 [DeviceTelemetry] Packet History - insert: Using new slot @uptime 65.449s TRACE NEW
DEBUG | ??:??:?? 65 [DeviceTelemetry] Use AES256 key!
DEBUG | ??:??:?? 65 [DeviceTelemetry] enqueue for send (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=48 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 65 [DeviceTelemetry] txGood=1,txRelay=0,rxGood=1,rxBad=0
DEBUG | ??:??:?? 65 [DeviceTelemetry] Use channel 0 (hash 0x6e)
DEBUG | ??:??:?? 65 [DeviceTelemetry] Use AES256 key!
DEBUG | ??:??:?? 65 [DeviceTelemetry] decoded message (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x0 Portnum=67 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 65 [Power] Battery: usbPower=1, isCharging=1, batMv=3985, batPct=79
DEBUG | ??:??:?? 65 [RadioIf] Started Tx (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=48 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 65 [RadioIf] Packet TX: 591ms
DEBUG | ??:??:?? 65 [RadioIf] 0 packets remain in the TX queue
DEBUG | ??:??:?? 66 [RadioIf] Completed sending (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=48 hopStart=6 relay=0x44 priority=10)
DEBUG | ??:??:?? 67 [RadioIf] Lora RX (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=48 rxSNR=6 rxRSSI=-45 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 67 [RadioIf] Packet RX: 591ms
DEBUG | ??:??:?? 67 [Router] Rx someone rebroadcasting for us (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=48 rxSNR=6 rxRSSI=-45 hopStart=6 relay=
DEBUG | ??:??:?? 67 [Router] Didn't find pending packet
INFO  | ??:??:?? 67 [Router] Packet History - insert: Reusing slot aged 1.693s TRACE MATCHED PACKET
DEBUG | ??:??:?? 67 [Router] Ignore dupe incoming msg (id=0x96654ea5 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=48 rxSNR=6 rxRSSI=-45 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 67 [Router] cancelSending id=0x96654ea5, removed=0
DEBUG | ??:??:?? 67 [Router] Incoming msg was filtered from 0xf5032f44
DEBUG | ??:??:?? 85 [Power] Battery: usbPower=1, isCharging=1, batMv=3988, batPct=79
**INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send: barometric_pressure=0.000000, current=0.000000, gas_resistance=0.000000, relative_humidity=0.000000, temperature=0.000000
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send: voltage=0.000000, IAQ=0, distance=0.000000, lux=0.000000
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send: wind speed=0.000000m/s, direction=0 degrees, weight=0.000000kg
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send: radiation=0.000000##R/h
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send: soil_temperature=26.000000, soil_moisture=0**
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] Partially randomized packet id 2367349414
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Send packet to mesh
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] Update DB node 0xf5032f44, rx_time=0
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Sort took 0 milliseconds
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] handleReceived(LOCAL) (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x0 Portnum=67 priority=70)     
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] No modules interested in portnum=67, src=LOCAL
INFO  | ??:??:?? 98 [EnvironmentTelemetry] Packet History - insert: Using new slot @uptime 98.350s TRACE NEW
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] Use AES256 key!
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] enqueue for send (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=38 hopStart=6 relay=0x44 priority=70)
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] txGood=2,txRelay=0,rxGood=2,rxBad=0
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] Use channel 0 (hash 0x6e)
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] Use AES256 key!
DEBUG | ??:??:?? 98 [EnvironmentTelemetry] decoded message (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x0 Portnum=67 hopStart=6 relay=0x44 priority=70)
DEBUG | ??:??:?? 98 [RadioIf] Started Tx (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=38 hopStart=6 relay=0x44 priority=70)
DEBUG | ??:??:?? 98 [RadioIf] Packet TX: 518ms
DEBUG | ??:??:?? 98 [RadioIf] 0 packets remain in the TX queue
DEBUG | ??:??:?? 99 [RadioIf] Completed sending (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=6 Ch=0x6e encrypted len=38 hopStart=6 relay=0x44 priority=70)
DEBUG | ??:??:?? 99 [RadioIf] Lora RX (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=38 rxSNR=4.75 rxRSSI=-75 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 99 [RadioIf] Packet RX: 518ms
DEBUG | ??:??:?? 99 [Router] Rx someone rebroadcasting for us (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=38 rxSNR=4.75 rxRSSI=-75 hopStart=6 rel
DEBUG | ??:??:?? 99 [Router] Didn't find pending packet
INFO  | ??:??:?? 99 [Router] Packet History - insert: Reusing slot aged 1.483s TRACE MATCHED PACKET
DEBUG | ??:??:?? 99 [Router] Ignore dupe incoming msg (id=0x8d1ae2a6 fr=0xf5032f44 to=0xffffffff, WantAck=0, HopLim=5 Ch=0x6e encrypted len=38 rxSNR=4.75 rxRSSI=-75 hopStart=6 relay=0xac)
DEBUG | ??:??:?? 99 [Router] cancelSending id=0x8d1ae2a6, removed=0

--- tried to look @ e. log here --

DEBUG | ??:??:?? 99 [Router] Incoming msg was filtered from 0xf5032f44
DEBUG | ??:??:?? 105 [Power] Battery: usbPower=1, isCharging=1, batMv=3989, batPct=79

Justin-Mann commented:
to be clear this is completely unmodified firmware and android client.  honestly is not bothering me.. but I figured I would let you guys know.

Justin-Mann commented:
also... wondering if u could reproduce because it WOULD bother me if this has to do with my firmware metrics (soil_temperature, soil_moisture) being included in the environmental telemetry protos (which I have a suspicion it might? not gonna jump 2 conclusions just yet.. gonna have a hot dog and potentially nap if you guys don't need me to do a thing.

DaneEvans commented:
It could well, but we need to move this convo to a different issue, it'll
get lost in here.

It's definitely a SW bug though, so write up an issue and include those
logs.

I'll try again to update some of my nodes to alpha. I wasn't able to
connect to them afterwards last weekend.

On Sat, 5 July 2025, 10:09 am Justin E. Mann, ***@***.***>
wrote:

> *Justin-Mann* left a comment (meshtastic/Meshtastic-Android#2227)
> <https://github.com/meshtastic/Meshtastic-Android/pull/2227#issuecomment-3037458389>
>
> also... wondering if u could reproduce because it WOULD bother me if this
> has to do with my firmware metrics (soil_temperature, soil_moisture) being
> included in the environmental telemetry protos (which I have a suspicion it
> might? not gonna jump 2 conclusions just yet.. gonna have a hot dog and
> potentially nap if you guys don't need me to do a thing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshtastic/Meshtastic-Android/pull/2227#issuecomment-3037458389>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIRN4PRTPTBTJTJCQKK2PT3G4JU7AVCNFSM6AAAAAB73V7HWGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAMZXGQ2TQMZYHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Justin-Mann commented:
![image](https://github.com/user-attachments/assets/6cfa669a-5623-4b3d-9770-575e3a4be786)
I have my doubts many folks r using hardware that would ever populate these atm.  no clients can currently take advantage of them.. This PR IS that.

DaneEvans commented:
But go eat first.  And maybe try to get onto the fdroid or other pre
release android builds.

We're doing 2 releases a week, and the play store is the slowest I think.

On Sat, 5 July 2025, 10:09 am Justin E. Mann, ***@***.***>
wrote:

> *Justin-Mann* left a comment (meshtastic/Meshtastic-Android#2227)
> <https://github.com/meshtastic/Meshtastic-Android/pull/2227#issuecomment-3037458389>
>
> also... wondering if u could reproduce because it WOULD bother me if this
> has to do with my firmware metrics (soil_temperature, soil_moisture) being
> included in the environmental telemetry protos (which I have a suspicion it
> might? not gonna jump 2 conclusions just yet.. gonna have a hot dog and
> potentially nap if you guys don't need me to do a thing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshtastic/Meshtastic-Android/pull/2227#issuecomment-3037458389>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIRN4PRTPTBTJTJCQKK2PT3G4JU7AVCNFSM6AAAAAB73V7HWGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAMZXGQ2TQMZYHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Justin-Mann commented:
bad thought = if this fixes it.. the ios client and web client probably need it too.

Justin-Mann commented:
we c

Justin-Mann commented:
![image](https://github.com/user-attachments/assets/df800cf5-ccc1-4cd4-aea3-02eb15cb033f)
just gonna put this here so u can c it.. maybe worth me running another scenario (remove GPS from RAK and do again).
I doubt this is an issue.. just causing my GPS not to fxn 2 the moment. the new WISblock GPS seems to do ok I think (from discord comments I have read recently.

Justin-Mann commented:
also a think I can make u aware of.. this is a lot, but here is HOW I'm making use of the firmware irl even without the android support this pr is intended to add.. https://www.instructables.com/RAKMeshtastic-Based-Irrigation-Monitor/  the firmware build I host for this is slightly modified to override the existing temp and relative humidity properties in the proto w/ my soil values.
suppose now is the time to be deliberate not in a hurry.. so I include everything I know or suspect for u.  I will do the report an issue thing after hotdogs.

DaneEvans commented:
Thanks Justin, 

The MR looks just about ready to go. 
- [ ] Is there native firmware support at the moment? 
    - [ ] which version? 
- [ ] Given that neither of us have the hardware to test it thoroughly, can we get a video with software generated on this branch,  firmware on 2-3 nodes (older before support, current with support (both with and without sensors)) and show us all of the pages you touched in the MR. 

I'll pull the details regarding this other bug out to #2348

Justin-Mann commented:
i did a thing.. [Bug]: Environment Log on Android causing crash #2349
:)
happy 4th man. u a gem.

Justin-Mann commented:
np

Justin-Mann commented:
I'll drop a vid tomorrow man.. not sure, but @thebentern may have the RAK hw comps necessary to reproduce as well.

Justin-Mann commented:
@DaneEvans @jamesarich 

video files and such uploaded to the Bug Report man.. I think that is what you were looking for?

I can flash with current stable firmware and (probably) show the log working as intended if you need it.. I have not done this, but I'm sure it works.

DaneEvans commented:
@Justin-Mann 
To get this into main, we need to stop changing things in it, even strings and such means we need to look over it again, and re-review it. 

It's waiting on the tasks in my comment from yesterday to ensure that it doesn't cause any collateral damage

Justin-Mann commented:
@DaneEvans I done man.. those were final submissions. they bothered me enough I did it anyways.  Once u told me your release schedule I made a decision.

Justin-Mann commented:
idk how to resolve conflict on protos here.. local branch works great when I test.

DaneEvans commented:
proto's are upstream changes. Just remove them from the MR, they shouldn't be commited at all 

We're still waiting on these to get this merged.
 - [ ] Is there native firmware support at the moment?
       - [ ] which version?
 - [ ] Given that neither of us have the hardware to test it thoroughly, can we get a video with software generated on this branch, firmware on 2-3 nodes (older before support, current with support (both with and without sensors)) and show us all of the pages you touched in the MR.

Justin-Mann commented:
will do.. when u posted that before I thought in regard 2 the bug report..
will grab videos and fix service example..

Justin-Mann commented:
Is there native firmware support at the moment?
- [2.6.13 & 2.7.1 (alphas) ] which version?

Justin-Mann commented:
back to the problem I ran into when I started messing w/ protos.. 

![image](https://github.com/user-attachments/assets/e012ba1d-7c62-4972-bfe7-137a758aff8d)

Not sure what I'm doing wrong.

I'll pick this up in AM... get recordings then too.

DaneEvans commented:
I fixed the protos in the branch, 
You'll need to let them update locally - just not commit them

Justin-Mann commented:
no. not sure if im even doing the right thing man.. I pulled origin.. let github desktop resolve conflicts.. still same issue going on.

Justin-Mann commented:
I hit it again in am.

Justin-Mann commented:
sure.

Justin-Mann commented:
rollback done. do u want me to update the base branch at this point? to resolve the channeloptions?

Justin-Mann commented:
I push everything here.. super broken.. not sure why.. I just begin again...
